sdk/python/kfp/client/client.py

```diff
@@ -1508,21 +1555,19 @@ def get_pipeline_version(
         """
         return self._pipelines_api.get_pipeline_version(version_id=version_id)
 
-    def delete_pipeline_version(self, version_id: str):
+    def delete_pipeline_version(self, version_id: str) -> None:
         """Deletes a pipeline version.
 
         Args:
             version_id: id of the pipeline version.
 
         Returns:
-            Object. If the method is called asynchronously, returns the request
-            thread.
+            None
 
         Raises:
             kfp_server_api.ApiException: If pipeline is not found.
         """
-        return self._pipelines_api.delete_pipeline_version(
-            version_id=version_id)
+        self._pipelines_api.delete_pipeline_version(version_id=version_id)
```

Comment:
@ji-yaqi, what do you think about this change to the return value in improve consistency across methods? This changes is in response to my comment here: https://github.com/kubeflow/pipelines/pull/7562#discussion_r850599235.

Currently, these methods return an empty dict `{}` which, like `None` is also falsey: `bool({}) == bool(None)`. We haven't been consistent with what we return between methods, so I suggested we return `None` from all of them. This seems like a good step toward consistency that, while not 100% backward compatible, is fairly safe. WDYT?

Reply:
Why are we changing the behavior to return null instead of whatever is returned previously?

Reply:
Currently, there is mixed behavior between analogous methods (the archive, unarchive, and delete methods). The Experiment API server always returns `{}`, but our client returns either `{}` or `None`. I was asking whether we should reconcile the inconsistencies.

However, upon closer look, the return values are not quite as inconsistent as I originally thought: the `archive_*` and `unarchive_*` methods return `None` while `delete_*` methods always return `{}`. 

Unless @chensun feels otherwise, can we stick with that pattern @droctothorpe? So your new `(un)archive_run` methods can return `None` and the `delete_run` method can return `{}`, leaving the existing methods unchanged.

Reply:
I rebased from master and I (think I) addressed the above notes. The only modification to the existing methods is the removal of `If the method is called asynchronously, returns the request thread.` from the doc strings, along with the suggested enhancements to the type hints. I had to update the `delete_run` doc string to specify that it returns `Object`, which you had expressed some justifiable reservations about earlier, but now it aligns with the existing delete methods. Let me know if you'd like me to modify that for any reason across the board. I also don't mind tackling that in a discrete PR if it makes sense. Otherwise this should be good to go.